langchain/src/util/sql_utils.ts

```diff
@@ -177,17 +177,28 @@ export const getTableAndColumnsName = async (
   }
 
   if (appDataSource.options.type === "mssql") {
-    sql =
+    const schema = appDataSource.options?.schema;
+    const sqlWithSchema =
       "SELECT " +
       "TABLE_NAME AS table_name, " +
       "COLUMN_NAME AS column_name, " +
       "DATA_TYPE AS data_type, " +
       "IS_NULLABLE AS is_nullable " +
       "FROM INFORMATION_SCHEMA.COLUMNS " +
+      `WHERE TABLE_SCHEMA = '${schema}'` +
       "ORDER BY TABLE_NAME, ORDINAL_POSITION;";
 
-    const rep = await appDataSource.query(sql);
+    const sqlWithoutSchema =
+      "SELECT " +
+      "TABLE_NAME AS table_name, " +
+      "COLUMN_NAME AS column_name, " +
+      "DATA_TYPE AS data_type, " +
+      "IS_NULLABLE AS is_nullable " +
+      "FROM INFORMATION_SCHEMA.COLUMNS " +
+      "ORDER BY TABLE_NAME, ORDINAL_POSITION;";
 
+    sql = schema ? sqlWithSchema : sqlWithoutSchema;
+    const rep = await appDataSource.query(sql);
```

Comment:
Drop `sqlWithoutSchema` and rename `sqlWithSchema` to just `sql`
```suggestion
    const rep = await appDataSource.query(sql);
```